File: update_config.py
#!/usr/bin/env python3
import json
import os

# Get the directory where this script is located
script_dir = os.path.dirname(os.path.abspath(__file__))
index_path = os.path.join(script_dir, "index.js")

# Path to the .claude.json file
config_path = os.path.expanduser("~/.claude.json")

# Read the existing config
with open(config_path, 'r') as f:
    config = json.load(f)

# Add the PDF reader MCP server to the global mcpServers
if "mcpServers" not in config:
    config["mcpServers"] = {}

config["mcpServers"]["pdf-reader"] = {
    "command": "node",
    "args": [index_path]
}

# Write back the updated config
with open(config_path, 'w') as f:
    json.dump(config, f, indent=2)

print("Successfully added pdf-reader MCP server to ~/.claude.json")
print(f"Server path: {index_path}")
print("\nMCP Servers configured:")
for server_name in config["mcpServers"]:
    print(f"  - {server_name}")

File: update-config.js
#!/usr/bin/env node

import { readFile, writeFile } from 'fs/promises';
import { homedir } from 'os';
import { join } from 'path';

async function updateConfig() {
  try {
    const configPath = join(homedir(), '.claude.json');

    // Read existing config
    const configData = await readFile(configPath, 'utf8');
    const config = JSON.parse(configData);

    // Add or update the PDF reader MCP server
    if (!config.mcpServers) {
      config.mcpServers = {};
    }

    config.mcpServers['pdf-reader'] = {
      command: 'npx',
      args: ['@hancengiz/pdf-reader-mcp-server']
    };

    // Write back the updated config
    await writeFile(configPath, JSON.stringify(config, null, 2), 'utf8');

    console.log('✅ Successfully updated pdf-reader MCP server in ~/.claude.json');
    console.log('\nConfiguration:');
    console.log('  command: npx');
    console.log('  args: @hancengiz/pdf-reader-mcp-server');
    console.log('\nMCP Servers configured:');
    for (const serverName in config.mcpServers) {
      console.log(`  - ${serverName}`);
    }
    console.log('\n⚠️  Please restart Claude Code for changes to take effect.');
  } catch (error) {
    console.error('Error updating config:', error.message);
    process.exit(1);
  }
}

updateConfig();
